tests/phpunit/tests/blocks/applyBlockHooksToContentFromPostObject.php

```diff
@@ -130,6 +140,33 @@ public function test_apply_block_hooks_to_content_from_post_object_respects_igno
 		$this->assertSame( $expected, $actual );
 	}
 
+	/**
+	 * @ticket 63287
+	 */
+	public function test_apply_block_hooks_to_content_from_post_object_does_not_insert_hooked_block_before_container_block() {
+		$filter = function ( $hooked_block_types, $relative_position, $anchor_block_type ) {
+			if ( 'core/post-content' === $anchor_block_type && 'before' === $relative_position ) {
+				$hooked_block_types[] = 'tests/dynamically-hooked-block-before-post-content';
```

Comment:
Where is the `tests/dynamically-hooked-block-before-post-content` block registered? I think I'm missing something here to understand the test.

Reply:
It isn't registered anywhere. It _might_ not be strictly needed -- if the test behaves as it should, i.e. it doesn't insert the block, then we shouldn't need to look up any more information from the block.

To be on the safe side, I've added registration (and unregistration during teardown): https://github.com/WordPress/wordpress-develop/pull/8700/commits/0c0e870fa84d1803912279cf9d69ac3a9f17f3e1

Reply:
Now, I get it. It check whether this block added throug the filter does not get injected 👍🏻